cli/src/ui.rs

```diff
@@ -329,10 +330,10 @@ impl PagerConfig {
             return Ok(Self::Disabled);
         };
         let args: CommandNameAndArgs = config.get("ui.pager")?;
-        if args.as_str() == Some(BUILTIN_PAGER_NAME) {
-            Ok(Self::Builtin(config.get("ui.streampager")?))
-        } else {
-            Ok(Self::External(args))
+        match (args.as_str(), env::var("PAGER")) {
+            (Some(BUILTIN_PAGER_NAME), _) => Ok(Self::Builtin(config.get("ui.streampager")?)),
+            (Some(ENV_PAGER_NAME), Ok(pager_env)) => Ok(Self::External((&pager_env).into())),
```

Comment:
":envpager" should be an error if `$PAGER` isn't set.

BTW, I personally don't think we need `:envpager`. Since the user has to configure it manually, they can just set their pager command as usual. I don't think it's common to temporarily override `$PAGER`. If they do, they can use `sh -c $PAGER` or something.

Reply:
The `sh` command is a good point. Apparently, this was already discussed in https://github.com/jj-vcs/jj/pull/7821#discussion_r2456005797.

I would still prefer having `:env`/`:envpager`. That `sh` command can take some quoting if you want to put it into a `jj config set` command and can be messed up by putting extra quotes around `$PAGER`, both mistakes resulting in subtle problems that may not be immediately apparent. It doesn't work on Windows.

But, if your or others' preference against it is strong enough, we'll probably manage without `:env` as well.